packages/sdk-auth/src/auth.js

```diff
@@ -237,10 +236,21 @@ export default class SdkAuth {
   }
 
   _getRequestConfig(config: CustomAuthOptions = {}): AuthOptions {
-    const mergedConfig = defaultsDeep({}, config, this.config)
 
-    // handle scopes array - defaultsDeep would merge arrays together
+    // It is important to note the difference between plain old javascript Object.assign()
+    // and the lodash _.defaultsDeep() functions especially how they are both used here
+
+    // _.defaultsDeep({ 'a': { 'b': 2 } }, { 'a': { 'b': 1, 'c': 3 } }) // { a: { b: 2, c: 3 } }
+    // Object.assign({ 'a': { 'b': 2 } }, { 'a': { 'b': 1, 'c': 3 } }) // { a: { b: 1, c: 3 } }
+
+    // handle scopes array - Object.assign would merge arrays together
```

Comment:
I think this is actually not really good if a missing dependency would change the behavior here. I would take a look what's the size of the config and do the merge straightforward in a function without the need of any dependency. Or directly use Object.assign